### PR TITLE
Add a job to check code updates

### PR DESCRIPTION
**Pull request by lzhecheng**
* add an Action job to check if generated code is up-to-date
* refactor update-mock-clients.sh
* generate mock code

Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
/ok-to-test

File: .github/workflows/test-updates.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+    - name: Set up Go 1.17
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
+    - name: Check-out code
+      uses: actions/checkout@v2
+    - name: Check updates
+      run: |
+        go install github.com/golang/mock/mockgen@v1.6.0
```
Comment (nilo19) [inline .github/workflows/test-updates.yml:18]: maybe we need to make it clear somewhere in the doc that we are using mockgen v1.6.0, as the generated contents may vary with different versions.

File: hack/verify-updates.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+make update
+diff="$(git diff)"
+
+if [ ! -z "$diff" ]; then
```
Comment (MartinForReal) [inline hack/verify-updates.sh:26]: Please refer to https://github.com/grawlinson/version-embedding/blob/806e1c43d5fde62f67ce14e5ddfce2fca428c329/Makefile#L16
I think `git status` is widely used in this scenario.

File: .github/workflows/test-updates.yml (hunk 1 above)
Comment (lzhecheng) [reply .github/workflows/test-updates.yml:18]: Sure. Added docs under `site/`

File: hack/verify-updates.sh (hunk 2 above)
Comment (lzhecheng) [reply hack/verify-updates.sh:26]: Good idea. Updated.

**Comment by MartinForReal** [comment]
I don't think this is a user facing change. Would you please change user facing doc to NONE?

File: .github/workflows/test-updates.yml (hunk 1 above)
Comment (MartinForReal) [reply .github/workflows/test-updates.yml:18]: I'm wondering if we can add a new make target to install mockgen and set it as a prerequisite for `make update`

Comment (lzhecheng) [reply .github/workflows/test-updates.yml:18]: I will say let's put install mockgen into `update-mock-clients.sh`. Install mockgen doesn't seem to be cloud-provider-azure specific target. What's your idea?

**Comment by lzhecheng** [comment]
> I don't think this is a user facing change. Would you please change user facing doc to NONE in pr description?

Sure.

File: site/content/en/development/github-actions.md/verify-code.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (nilo19) [inline site/content/en/development/github-actions.md/verify-code.md:1]: could you place it under https://kubernetes-sigs.github.io/cloud-provider-azure/development/dependencies/?

Comment (lzhecheng) [reply site/content/en/development/github-actions.md/verify-code.md]: done

**Pull request merged**